scripts/client/lint-rust.mjs

```diff
@@ -10,6 +10,7 @@ import {
 // Configure arguments here.
 const lintArgs = [
   '-Zunstable-options',
+  '--all-targets',
```

Comment:
I wanted to add `--all-features` here as well, but looks like maybe we're not configuring `kaigan` serde support properly?

```
the trait `generated::accounts::address_lookup_table::_::_serde::Deserialize<'_>` is not implemented for `kaigan::types::U64PrefixVec<solana_program::pubkey::Pubkey>`
```

Reply:
Nvm, I added the fix for this. But I found another clippy error for the Anchor one.

```
expected `anchor_lang::prelude::Pubkey`, found `solana_program::pubkey::Pubkey`
   |                   help: change the output type to match the trait: `anchor_lang::prelude::Pubkey`
```

Reply:
We might need to adjust the renderer @lorisleiva so that it uses more Anchor-compatible types, or can somehow tell rustc that the types are the same. Maybe it's a versioning thing?

Reply:
Sorry, I'm not sure I understand what you'd like to adjust. Could you share the bit of code that is causing this clippy error?

Reply:
It's a clippy error that arises if you pass `--all-features`, which runs lint on the `anchor` feature.

Looks like it's because Anchor 0.30 runs `solana-program` v1.16.
https://github.com/coral-xyz/anchor/blob/852fcc77beb6302474a11e0f8e6f1e688021be36/lang/Cargo.toml#L60

Reply:
So I guess we just have to make sure create-solana-program generates a template with compatible SDK versions for Anchor, or remove the Anchor support from the base template.

Reply:
Oh I see what you mean.

The way `create-solana-program` figures out versioning is by checking the version of your local binaries. So if you end up with incompatible versions, it should mean that the versions you had locally were incompatible at the time of generating the repo.

Recently, we added [a new option on the Rust renderer](https://github.com/codama-idl/codama/pull/250) that turns off any Anchor-related traits so you don't need to have any dependency on Anchor. So maybe we should use that here.